quartz/util/theme.ts

```diff
@@ -15,11 +15,19 @@ interface Colors {
   darkMode: ColorScheme
 }
 
+interface FontInfo {
+  /**
+   * e.g. "ital,wght@0,400;1,200"
+   */
+  features: string
+}
```

Comment:
If you're going to separate the fonts from the features, why not use multiple values for the different options? Like instead of just `features`, maybe consider `mode` for italics/bold/default and `weight` for the font weight.

Reply:
This is reasonable and I thought about it, but
- if Google Fonts changes their API, this code would need to be updated quickly,
- making it as expressive as the string while keeping a good API would be hard,
- specifying the particular weights is something that only people who are already probably heavily editing the theme are going to think about, and those people are probably already familiar with the Google Fonts API
so I decided that keeping it a string would be less of a maintenance burden and less complex in most situations.

Reply:
> This is reasonable and I thought about it, but
> - if Google Fonts changes their API, this code would need to be updated quickly,
> - making it as expressive as the string while keeping a good API would be hard,
> - specifying the particular weights is something that only people who are already probably heavily editing the theme are going to think about, and those people are probably already familiar with the Google Fonts API
> so I decided that keeping it a string would be less of a maintenance burden and less complex in most situations.
> 
> (If it seems like most people would prefer this be broken out into multiple options anyways, I can update it to do that / give people both options with `features` winning out.)

I feel like user that want to heavily customize their fonts would do so in `custom.scss` anyway.

Reply:
My issue with that is that the current method of specifying fonts outputs a <link> to Google Fonts at the top - I want to be able to modify that in a more advanced way. While I could just put an `@import` in my custom.scss, if the Google Fonts <link> element is incorrect, that's another request to a resource that might not actually be used / be what I want.

Reply:
> My issue with that is that the current method of specifying fonts outputs a <link> to Google Fonts at the top - I want to be able to modify that in a more advanced way. While I could just put an `@import` in my custom.scss, if the Google Fonts <link> element is incorrect, that's another request to a resource that might not actually be used / be what I want.

In that case, you could just change the font source in your `quartz.config.ts` to local and set the desired font in your `custom.scss` through `@font-face`, be it a Google font or not.

I suppose I'm kinda looking for what problem is solved with the proposed change. The current configuration makes it easy for inexperienced users to configure a Google-font. Advanced users would likely use CSS. I'm unsure what group is better served by the proposed changes.

Reply:
I think for cases like custom fonts, users best to use `local` and setup `@font-face` in custom.scss

Reply:
no need to add unnecessary abstraction on top